data/txHandler.go

```diff
@@ -313,9 +308,8 @@ func (handler *TxHandler) backlogWorker() {
 				transactionMessagesDroppedFromBacklog.Inc(nil)
 				return
 			}
-			if wi.capguard != nil {
-				wi.capguard.Served()
-			}
+			// mark the capguard as served
+			handleCapguard(wi.capguard, false, true)
```

Comment:
This call is not needed. The case for which this is called, can be covered with the previous call. 
```suggestion
```